middlewares/zap_logger.go

```diff
@@ -2,55 +2,54 @@ package middlewares
 
 import (
 	"fmt"
+	"log"
 	"time"
 
 	"github.com/labstack/echo/v4"
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
 )
 
-func ZapLogger(log *zap.Logger) echo.MiddlewareFunc {
-	return func(next echo.HandlerFunc) echo.HandlerFunc {
-		return func(c echo.Context) error {
-			start := time.Now()
-
-			err := next(c)
-			if err != nil {
-				c.Error(err)
-			}
-
-			req := c.Request()
-			res := c.Response()
-
-			fields := []zapcore.Field{
-				zap.String("remote_ip", c.RealIP()),
-				zap.String("latency", time.Since(start).String()),
-				zap.String("host", req.Host),
-				zap.String("request", fmt.Sprintf("%s %s", req.Method, req.RequestURI)),
-				zap.Int("status", res.Status),
-				zap.Int64("size", res.Size),
-				zap.String("user_agent", req.UserAgent()),
-			}
-
-			id := req.Header.Get(echo.HeaderXRequestID)
-			if id == "" {
-				id = res.Header().Get(echo.HeaderXRequestID)
-			}
-			fields = append(fields, zap.String("request_id", id))
-
-			n := res.Status
-			switch {
-			case n >= 500:
-				log.With(zap.Error(err)).Error("Server error", fields...)
-			case n >= 400:
-				log.With(zap.Error(err)).Warn("Client error", fields...)
-			case n >= 300:
-				log.Info("Redirection", fields...)
-			default:
-				log.Info("Success", fields...)
-			}
-
-			return nil
+func ZapLogger(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		start := time.Now()
+
+		err := next(c)
+		if err != nil {
+			c.Error(err)
+		}
+
+		req := c.Request()
+		res := c.Response()
+
+		// Logging information
+		remoteIP := c.RealIP()
+		latency := time.Since(start)
+		host := req.Host
+		request := fmt.Sprintf("%s %s", req.Method, req.RequestURI)
+		status := res.Status
+		size := res.Size
+		userAgent := req.UserAgent()
+
+		id := req.Header.Get(echo.HeaderXRequestID)
+		if id == "" {
+			id = res.Header().Get(echo.HeaderXRequestID)
 		}
+
+		// Log messages based on status code
+		switch {
+		case status >= 500:
+			log.Printf("Server error: %s - Remote IP: %s, Latency: %s, Host: %s, Request: %s, Status: %d, Size: %d, User Agent: %s, Request ID: %s",
+				err.Error(), remoteIP, latency, host, request, status, size, userAgent, id)
+		case status >= 400:
+			log.Printf("Client error: %s - Remote IP: %s, Latency: %s, Host: %s, Request: %s, Status: %d, Size: %d, User Agent: %s, Request ID: %s",
+				err.Error(), remoteIP, latency, host, request, status, size, userAgent, id)
+		case status >= 300:
+			log.Printf("Redirection - Remote IP: %s, Latency: %s, Host: %s, Request: %s, Status: %d, Size: %d, User Agent: %s, Request ID: %s",
+				remoteIP, latency, host, request, status, size, userAgent, id)
+		default:
+			log.Printf("Success - Remote IP: %s, Latency: %s, Host: %s, Request: %s, Status: %d, Size: %d, User Agent: %s, Request ID: %s",
+				remoteIP, latency, host, request, status, size, userAgent, id)
+		}
+
+		return nil
```

Comment:
No need to change logger middleware, that's already doing what you've done in a more clear way.

Reply:
fixed